titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/event/JobEventFactory.java

```diff
@@ -82,7 +88,20 @@ public JobManagerReconcilerEvent newModelUpdateErrorEvent(ReconciliationEngine<J
                                                               EntityHolder previousEntityHolder,
                                                               Throwable error,
                                                               String transactionId) {
-        return new JobModelReconcilerEvent.JobModelUpdateErrorReconcilerEvent(getJob(engine, modelActionHolder), (TitusChangeAction) changeAction, modelActionHolder, previousEntityHolder, error, transactionId);
+        return new JobModelReconcilerEvent.JobModelUpdateErrorReconcilerEvent(getJob(engine, modelActionHolder), (TitusChangeAction) changeAction, modelActionHolder, previousEntityHolder, error, transactionId,
+                (CallMetadata)getEntityHolder(engine, modelActionHolder).getAttributes().get(JobAttributes.JOB_ATTRIBUTE_CALLMETADATA));
+    }
+
+    private EntityHolder getEntityHolder(ReconciliationEngine<JobManagerReconcilerEvent> engine, ModelActionHolder modelActionHolder) {
```

Comment:
We should replace this function with a function returning `CallMetadata` value fetched from the `EntityHolder`.
If the value is null, we should return a marker value (`UNDEFINED_CALL_METADATA`?)